rgsync/Connectors/redis_connector.py

```diff
@@ -0,0 +1,178 @@
+from rgsync.common import *
+from redisgears import getMyHashTag as hashtag
+
+
+# redis connection class
+class RedisConnection():
+    def __init__(self, host, port, password=None):
+        self._host = host
+        self._port = port
+        self._password = password
+
+    @property
+    def host(self):
+        return self._host() if callable(self._host) else self._host
+
+    @property
+    def port(self):
+        return self._port() if callable(self._port) else self._port
+
+    @property
+    def password(self):
+        return self._password() if callable(self._password) else self._password
+
+    def Connect(self):
+        from redis.client import Redis
+        try:
+            WriteBehindLog("Connect: connecting to {}:{} password: {}".format(self.host, self.port, self.password))
+            r = Redis(host=self.host, port=self.port, password=self.password, decode_responses=True)
+        except Exception as e:
+            msg = "Cannot connect to Redis. Exception: {}".format(e)
+            WriteBehindLog(msg)
+            raise Exception(msg) from None
+        return r
+
+
+# redis cluster connection class
+class RedisClusterConnection():
+    def __init__(self, host=None, port=None, cluster_nodes=None, password=None):
```

Comment:
By the way, why do you need the host and the port here and not just the cluster_nodes?

Reply:
@MeirShpilraien 
https://github.com/Grokzen/redis-py-cluster/blob/73f27edf7ceb4a408b3008ef7d82dac570ab9c6a/rediscluster/client.py#L303

RedisCluster class provides option to pass single or multiple nodes of command as parameter based on this I have used the same.